src/Elastic.Apm/Api/ITracer.cs

```diff
@@ -22,6 +22,38 @@ public interface ITracer
 		/// </summary>
 		ITransaction CurrentTransaction { get; }
 
+		/// <summary>
+		/// Captures a custom error and reports it to the APM server.
+		/// </summary>
+		/// <param name="message">The error message.</param>
+		/// <param name="culprit">The culprit of the error.</param>
+		/// <param name="frames">The stack trace when the error occured.</param>
+		/// <param name="parentId">
+		/// The parent ID that is attached to the error. In case it's null the parent
+		/// will be automatically set to the current instance
+		/// </param>
+		void CaptureError(string message, string culprit, StackFrame[] frames = null, string parentId = null);
+
+		/// <summary>
+		/// Captures an exception and reports it to the APM server.
+		/// </summary>
+		/// <param name="exception">The exception to capture.</param>
+		/// <param name="culprit">The value of this parameter is shown as 'Culprit' on the APM UI.</param>
+		/// <param name="isHandled">Indicates whether the exception is handled or not.</param>
+		/// <param name="parentId">
+		/// The parent ID that is attached to the error. In case it's null the parent
+		/// will be automatically set to the current instance
+		/// </param>
+		void CaptureException(Exception exception, string culprit = null, bool isHandled = false, string parentId = null);
+
+		/// <summary>
+		/// Captures a log line as an APM error.
+		/// </summary>
+		/// <param name="logOnError"> The logline itself </param>
+		/// <param name="parentId"> ParentId pointing to the parent transaction or span. </param>
+		/// <param name="exception"> Exception which was captured as part of the log. </param>
+		void CaptureLogAsError(ErrorLog logOnError, string parentId = null, Exception exception = null);
```

Comment:
What are your thoughts here about introducing an overload to `CaptureError` for this? For example

```c#
void CaptureError(ErrorLog errorLog, string parentId = null, Exception exception = null);
```

Capturing an error log, or message and culprit (and capturing an exception too) all map to the notion of an APM error, which could be surfaced in the API with an overloaded method.

Reply:
It's a reasonable approach, but I don't see a huge difference.

We already have `CaptureError`, `CaptureException`, and now we'd introduce `CaptureLogAsError`. With this I think the method name describes better what the intention is. I don't think using overloads would be very different.